mypyc/codegen/emitwrapper.py

```diff
@@ -278,29 +278,28 @@ def generate_bin_op_wrapper(cl: ClassIR, fn: FuncIR, emitter: Emitter) -> str:
 
     Both arguments and the return value are PyObject *.
     """
-    gen = WrapperGenerator(cl, emitter)
+    gen = WrapperGenerator(cl.name_prefix(emitter.names), emitter)
```

Comment:
(In conjunction with my other comment) In particular this prevents caller from doing something like `WrapperGenerator(cl.name_prefix(emitter2.names), emitter)`

Reply:
Fixed!